agent/pkg/builder/container/container_builder.go

```diff
@@ -400,41 +402,54 @@ func prepareImage(dc *DockerContainerBuilder) error {
 	return nil
 }
 
-func createNetwork(dc *DockerContainerBuilder) *string {
-	networkName := fmt.Sprintf("%s-network", dc.networkMode)
+func createNetworks(dc *DockerContainerBuilder) []string {
+	var networkIds []string
 
-	filter := filters.NewArgs()
-	filter.Add("name", networkName)
+	for _, network := range dc.networks {
+		networkName := network
 
-	networks, err := dc.client.NetworkList(dc.ctx, types.NetworkListOptions{Filters: filter})
-	if err != nil {
-		logWrite(dc, fmt.Sprintln("Failed to create network: ", err.Error()))
-		return nil
-	}
+		filter := filters.NewArgs()
+		filter.Add("name", fmt.Sprintf("^%s$", networkName))
+		networks, err := dc.client.NetworkList(dc.ctx, types.NetworkListOptions{Filters: filter})
+		if err != nil {
+			logWrite(dc, fmt.Sprintln("Failed to create network: ", err.Error()))
+			return nil
+		}
 
-	if len(networks) == 1 {
-		return &networks[0].ID
-	}
+		if len(networks) == 1 {
+			networkIds = append(networkIds, networks[0].ID)
+			continue
+		}
 
-	networkOpts := types.NetworkCreate{
-		CheckDuplicate: true,
-	}
-	networkResult, err := dc.client.NetworkCreate(dc.ctx, networkName, networkOpts)
-	if err != nil {
-		logWrite(dc, fmt.Sprintln("Failed to create network: ", err.Error()))
-		return nil
+		networkOpts := types.NetworkCreate{
+			CheckDuplicate: true,
+		}
+
+		networkResult, err := dc.client.NetworkCreate(dc.ctx, networkName, networkOpts)
+		if err != nil {
+			logWrite(dc, fmt.Sprintln("Failed to create network: ", err.Error()))
+			return nil
+		}
+
+		networkID := networkResult.ID
+		networkIds = append(networkIds, networkID)
```

Comment:
Could be simplyfied something like `networkIds = append(networkIds, networkResult.ID)`